source/server/session_resource_repository.h

```diff
@@ -74,15 +76,17 @@ class SessionResourceRepository {
   };
 
   SessionRepository* session_repository_;
-  ResourceMap resource_map_;
+  // shared_ptr to allow sharing ownership with cleanup delegate in register_dependent_session.
```

Comment:
Note: my test for this change uncovered a bug in the original design. The dependent sessions can get removed after resource_repository is cleaned up (the session_repository lives longer). Making this a shared_ptr lets us pass off shared ownership to the cleanup delegate so this can still be cleaned up later.

Reply:
Actually, this is a mess. I'm planning to stick with the general idea of excluding cross-driver sessions from the revere lookup. But the notifications/synchronization for cross-driver sessions in general are in worse shape than I thought.

Reply:
Update. I'm OK with the state as of the last couple commits. So, please review as normal.